automation_tools/satellite6/capsule.py

```diff
@@ -0,0 +1,398 @@
+        self.content_view = data.get('content-view')
+        self.activation_key = data.get('activation-key')
+        self.admin_user = data.get('admin-user')
+        self.admin_password = data.get('admin-password')
+        defaults = data.get('defaults')
+        if defaults is not None and isinstance(defaults, dict):
+            key_filename = defaults.get('key-filename')
+            self._key_filenames.add(key_filename)
+            self.defaults = Credentials(
+                user=defaults.get('user'),
+                password=defaults.get('password'),
+                key_filename=key_filename,
+            )
+        server = data.get('server')
+        if server is not None and isinstance(server, dict):
+            key_filename = server.get('key-filename')
+            self._key_filenames.add(key_filename)
+            self.server = HostConfig(
+                hostname=server.get('hostname'),
+                user=server.get('user', self.defaults.user),
+                password=server.get('password', self.defaults.password),
+                key_filename=key_filename,
+            )
+        capsules = data.get('capsules')
+        if capsules is not None and isinstance(capsules, list):
+            for capsule in capsules:
+                if capsule is not None and isinstance(capsule, dict):
+                    key_filename = capsule.get('key-filename')
+                    self._key_filenames.add(key_filename)
+                    self.capsules.append(HostConfig(
+                        hostname=capsule.get('hostname'),
+                        user=capsule.get('user', self.defaults.user),
+                        password=capsule.get(
+                            'password', self.defaults.password),
+                        key_filename=key_filename,
+                    ))
+
+    @property
+    def key_filenames(self):
+        """Return a list of collect key filenames or None if the list is
+        empty.
+
+        """
+        if self._key_filenames:
+            return list(self._key_filenames)
+        else:
+            return None
+
+    @property
+    def passwords(self):
+        """Return a dict in the format suited for Fabric usage in order to
+        define passwords for hosts.
+
+        """
+        passwords = {}
+        if self.server.password and not self.server.key_filename:
+            passwords[self.server.host_string] = self.server.password
+        for capsule in self.capsules:
+            if capsule.password and not capsule.key_filename:
+                passwords[capsule.host_string] = capsule.password
+        return passwords
+
+
+def _get_config():
+    """Get the capsule configuration if available in the fabric environment
+    else raise ``ImproperlyConfigured``.
+
+    """
+    config = env.get('capsule_config')
+    if config is None:
+        raise ImproperlyConfigured(
+            'Make sure to run load_capsule_config task.')
+    return config
```

Comment:
Good approach, but will leave as is because there are other places in automation-tools that uses similar approach.